docs/content/ingestion/native_tasks.md

```diff
@@ -0,0 +1,508 @@
+|maxRowsInMemory|Used in determining when intermediate persists to disk should occur. Normally user does not need to set this, but depending on the nature of data, if rows are short in terms of bytes, user may not want to store a million rows in memory and this value should be set.|1000000|no|
+|maxBytesInMemory|Used in determining when intermediate persists to disk should occur. Normally this is computed internally and user does not need to set it. This value represents number of bytes to aggregate in heap memory before persisting. This is based on a rough estimate of memory usage and not actual usage. The maximum heap memory usage for indexing is maxBytesInMemory * (2 + maxPendingPersists)|1/6 of max JVM memory|no|
+|maxTotalRows|Total number of rows in segments waiting for being published. Used in determining when intermediate publish should occur.|150000|no|
+|numShards|Directly specify the number of shards to create. If this is specified and 'intervals' is specified in the granularitySpec, the index task can skip the determine intervals/partitions pass through the data. numShards cannot be specified if targetPartitionSize is set.|null|no|
+|indexSpec|defines segment storage format options to be used at indexing time, see [IndexSpec](#indexspec)|null|no|
+|maxPendingPersists|Maximum number of persists that can be pending but not started. If this limit would be exceeded by a new intermediate persist, ingestion will block until the currently-running persist finishes. Maximum heap memory usage for indexing scales with maxRowsInMemory * (2 + maxPendingPersists).|0 (meaning one persist can be running concurrently with ingestion, and none can be queued up)|no|
+|forceExtendableShardSpecs|Forces use of extendable shardSpecs. Experimental feature intended for use with the [Kafka indexing service extension](../development/extensions-core/kafka-ingestion.html).|false|no|
+|reportParseExceptions|If true, exceptions encountered during parsing will be thrown and will halt ingestion; if false, unparseable rows and fields will be skipped.|false|no|
+|pushTimeout|Milliseconds to wait for pushing segments. It must be >= 0, where 0 means to wait forever.|0|no|
+|segmentWriteOutMediumFactory|Segment write-out medium to use when creating segments. See [Indexing Service Configuration](../configuration/indexing-service.html) page, "SegmentWriteOutMediumFactory" section for explanation and available options.|Not specified, the value from `druid.peon.defaultSegmentWriteOutMediumFactory` is used|no|
+|maxNumSubTasks|Maximum number of tasks which can be run at the same time.|Integer.MAX_VALUE|no|
+|maxRetry|Maximum number of retries on task failures.|3|no|
+|taskStatusCheckPeriodMs|Polling period in milleseconds to check running task statuses.|1000|no|
+|chatHandlerTimeout|Timeout for reporting the pushed segments in worker tasks.|PT10S|no|
+|chatHandlerNumRetries|Retries for reporting the pushed segments in worker tasks.|5|no|
+
+#### HTTP Endpoints
```

Comment:
Is there any relation in this "per peon API endpoints" and the coordinator discovery stuff that is being developed? specifically, @himanshug  and others are trying to push the coordinator to be able to assist in discovering cluster related resources. Is there any plans to tie in peon discovery or peon service discovery into the coordinator discovery items as well?

Reply:
I don't think coordinators are responsible for discovering/managing peons. In Druid, all modules are designed to work independently. So, It should be done in the indexing service module, especially by overlords.

Reply:
There's a push by many to see if the coordinator and the overlord can be the same node, so be aware that the functionality of one might be combined into the other before too long.

For us internally we do not combine them because the coordinators have to be pretty beefy to handle the qty of segments they have, but for smaller clusters having them as one node is often desired and way easier to manage.

And reducing the load on zookeeper is always desirable if possible, so I'm curious if any thought has been given for how per peon API discovery stuff should or could be handled

Reply:
Well, I haven't thought about that. For now, I would advocate for keeping overlords and coordinators separately unless there is a critical problem and a good alternative design to solve that. I think simply combining the coordinator and the overlord isn't a good solution because, as you said, it's not appropriate for large clusters as yours.